web/src/components/apps/Dashboard.tsx

```diff
@@ -26,47 +26,136 @@ const COMMON_ERRORS = {
   "no such host": "No such host",
 };
 
-class Dashboard extends Component {
-  state = {
-    appName: "",
-    iconUri: "",
-    currentVersion: {},
-    downstream: [],
-    links: [],
-    checkingForUpdates: false,
-    checkingUpdateMessage: "Checking for updates",
-    checkingForUpdateError: false,
-    appLicense: null,
-    activeChart: null,
-    crosshairValues: [],
-    noUpdatesAvalable: false,
-    updateChecker: new Repeater(),
-    uploadingAirgapFile: false,
-    airgapUploadError: null,
-    viewAirgapUploadError: false,
-    viewAirgapUpdateError: false,
-    airgapUpdateError: "",
-    startSnapshotErrorMsg: "",
-    showAutomaticUpdatesModal: false,
-    showAppStatusModal: false,
-    dashboard: {
-      appStatus: null,
-      metrics: [],
-      prometheusAddress: "",
-    },
-    getAppDashboardJob: new Repeater(),
-    fetchAppDownstreamJob: new Repeater(),
-    gettingAppLicenseErrMsg: "",
-    startSnapshotOptions: [
-      { option: "partial", name: "Start a Partial snapshot" },
-      { option: "full", name: "Start a Full snapshot" },
-      { option: "learn", name: "Learn about the difference" },
-    ],
-    selectedSnapshotOption: { option: "full", name: "Start a Full snapshot" },
-    snapshotDifferencesModal: false,
+// Types
+import {
+  App,
+  AppLicense,
+  Downstream,
+  DashboardResponse,
+  DashboardActionLink,
+  KotsParams,
+  ResourceStates,
+  Version,
+} from "@types";
```

Comment:
see comment at the bottom about importing from types and using interfaces instead!